v0/destinations/hs/HSTransform-v1.js

```diff
@@ -0,0 +1,350 @@
+      const { message, metadata, destination } = event;
+      const endpoint = get(message, "endpoint");
+
+      const batchedResponse = defaultBatchRequestConfig();
+      batchedResponse.batchedRequest.headers = message.headers;
+      batchedResponse.batchedRequest.endpoint = endpoint;
+      batchedResponse.batchedRequest.body = message.body;
+      batchedResponse.batchedRequest.params = message.params;
+      batchedResponse.batchedRequest.method =
+        defaultGetRequestConfig.requestMethod;
+      batchedResponse.metadata = [metadata];
+      batchedResponse.destination = destination;
+
+      trackResponseList.push(
+        getSuccessRespEvents(
+          batchedResponse.batchedRequest,
+          batchedResponse.metadata,
+          batchedResponse.destination
+        )
+      );
+    } else {
+      // making chunks for identify
+      eventsChunk.push(event);
+    }
+  });
+
+  // eventChunks = [[e1,e2,e3,..batchSize],[e1,e2,e3,..batchSize]..]
+  const arrayChunksIdentify = _.chunk(eventsChunk, MAX_BATCH_SIZE);
+
+  // list of chunks [ [..], [..] ]
+  arrayChunksIdentify.forEach(chunk => {
+    const identifyResponseList = [];
+    const metadata = [];
+
+    // extracting destination, apiKey value
+    // from the first event in a batch
+    const { destination } = chunk[0];
+    const { Config } = destination;
+
+    let batchEventResponse = defaultBatchRequestConfig();
+
+    chunk.forEach(ev => {
+      // if source is of rETL
+      if (ev.message.source === "rETL") {
+        identifyResponseList.push({ ...ev.message.body.JSON });
+        batchEventResponse.batchedRequest.body.JSON = {
+          inputs: identifyResponseList
+        };
+        batchEventResponse.batchedRequest.endpoint = `${ev.message.endpoint}/batch/create`;
+        metadata.push(ev.metadata);
+      } else {
+        const { email, updatedProperties } = getEmailAndUpdatedProps(
+          ev.message.body.JSON.properties
+        );
+        // eslint-disable-next-line no-param-reassign
+        ev.message.body.JSON.properties = updatedProperties;
+        identifyResponseList.push({
+          email,
+          properties: ev.message.body.JSON.properties
+        });
+        metadata.push(ev.metadata);
+        batchEventResponse.batchedRequest.body.JSON_ARRAY = {
+          batch: JSON.stringify(identifyResponseList)
+        };
+        batchEventResponse.batchedRequest.endpoint = BATCH_CONTACT_ENDPOINT;
+      }
+    });
+
+    batchEventResponse.batchedRequest.headers = {
+      "Content-Type": "application/json"
+    };
+
+    // choosing API Type
+    if (Config.authorizationType === "newPrivateAppApi") {
+      // Private Apps
+      batchEventResponse.batchedRequest.headers = {
+        ...batchEventResponse.batchedRequest.headers,
+        Authorization: `Bearer ${Config.accessToken}`
+      };
+    } else {
+      // API Key
+      batchEventResponse.batchedRequest.params = { hapikey: Config.apiKey };
+    }
```

Comment:
We can do this out side the chunk loop don't see any dependency to do within the loop